src/server/req_jobobit.c

```diff
@@ -1761,6 +1761,7 @@ job_obit(struct resc_used_update *pruu, int stream)
 			/* tell mom to trash job		    */
 			DBPRT(("%s: job %s not in exiting state!\n",
 				__func__, pruu->ru_pjobid))
+			pjob->ji_discarding = 0;
```

Comment:
If the job got re-queued while mom is down, it will set the discard flag and proceed. The job cannot run as long as it carries the flag. When the mom comes up it will send a job-obit and the server will reject it by sending bad-obit as the job already got re-queued. MoM will then discard the job upon receiving the bad-obit. Server needs to clear the discard flag before sending the bad-obit so the job can be run.

Reply:
got it, but still confirming: this is the case when mgr does force qrun/qrerun of job and node is down, right?

Reply:
yes. you are right!